protocol/x/accountplus/authenticator/authentication_request.go

```diff
@@ -0,0 +1,284 @@
+	// Retrieve and build the signer data struct
+	baseAccount := ak.GetAccount(ctx, account)
+	genesis := ctx.BlockHeight() == 0
+	chainID := ctx.ChainID()
+	var accNum uint64
+	if !genesis {
+		accNum = baseAccount.GetAccountNumber()
+	}
+	var sequence uint64
+	if baseAccount != nil {
+		sequence = baseAccount.GetSequence()
+	}
+
+	return authsigning.SignerData{
+		ChainID:       chainID,
+		AccountNumber: accNum,
+		Sequence:      sequence,
+	}
+}
+
+// extractExplicitTxData makes the transaction data concrete for the authentication request. This is necessary to
+// pass the parsed data to the cosmwasm authenticator.
+func extractExplicitTxData(tx sdk.Tx, signerData authsigning.SignerData) (ExplicitTxData, error) {
+	timeoutTx, ok := tx.(sdk.TxWithTimeoutHeight)
+	if !ok {
+		return ExplicitTxData{}, errorsmod.Wrap(sdkerrors.ErrInvalidType, "failed to cast tx to TxWithTimeoutHeight")
+	}
+	memoTx, ok := tx.(sdk.TxWithMemo)
+	if !ok {
+		return ExplicitTxData{}, errorsmod.Wrap(sdkerrors.ErrInvalidType, "failed to cast tx to TxWithMemo")
+	}
+
+	// Encode messages as Anys and manually convert them to a struct we can serialize to json for cosmwasm.
+	txMsgs := tx.GetMsgs()
+	msgs := make([]LocalAny, len(txMsgs))
+	for i, txMsg := range txMsgs {
+		encodedMsg, err := types.NewAnyWithValue(txMsg)
+		if err != nil {
+			return ExplicitTxData{}, errorsmod.Wrap(err, "failed to encode msg")
+		}
+		msgs[i] = LocalAny{
+			TypeURL: encodedMsg.TypeUrl,
+			Value:   encodedMsg.Value,
+		}
+	}
+
+	return ExplicitTxData{
+		ChainID:         signerData.ChainID,
+		AccountNumber:   signerData.AccountNumber,
+		AccountSequence: signerData.Sequence,
+		TimeoutHeight:   timeoutTx.GetTimeoutHeight(),
+		Msgs:            msgs,
+		Memo:            memoTx.GetMemo(),
+	}, nil
+}
+
+// extractSignatures returns the signature data for each signature in the transaction and
+// the one for the current signer.
+//
+// This function also checks for replay attacks. The replay protection needs to be able to match the signature to the
+// corresponding signer, which involves iterating over the signatures. To avoid iterating over the signatures twice,
+// we do replay protection here instead of in a separate replay protection function.
+//
+// Only SingleSignatureData is supported. Multisigs can be implemented by using partitioned compound authenticators
+func extractSignatures(
+	txSigners []sdk.AccAddress,
+	txSignatures []signing.SignatureV2,
+	account sdk.AccAddress,
+) (signatures [][]byte, msgSignature []byte, err error) {
+	for i, signature := range txSignatures {
+		single, ok := signature.Data.(*signing.SingleSignatureData)
+		if !ok {
+			return nil,
+				nil,
+				errorsmod.Wrap(
+					sdkerrors.ErrInvalidType,
+					"failed to cast signature to SingleSignatureData",
+				)
+		}
+
+		signatures = append(signatures, single.Signature)
+
+		if txSigners[i].Equals(account) {
+			msgSignature = single.Signature
+		}
```

Comment:
Note that I am skipping ReplayProtection [here](https://github.com/osmosis-labs/osmosis/blob/main/x/smart-account/authenticator/authentication_request.go#L167). 

Osmosis's implementation only compares sequence numbers. since we have our own timestamp nonce logic, probably better to leave it out of the smart account's authenticator flows

Reply:
cc @teddyding

Reply:
Does `GenerateAuthenticationRequest` flow get invoked in another antehandler? Is the regular `sigVerficationHandler` bypassed when this authenticator is in effect? I assume yes since the regular `sigVerificationHandler` doesn't understand the alternative authenticator/